docker/pulsar/Dockerfile

```diff
@@ -59,16 +59,16 @@ RUN apk add --no-cache \
        musl-dev \
        libffi-dev \
        py3-pip \
-       py3-grpcio \
```

Comment:
since `py3-grpcio` is already installed, why isn't that used? isn't there a way to prevent upgrading of the grpcio package?

Reply:
I'm not good at pip, and as far as I know there's no way. The pip will uninstall the installed, and then to install a new grpcio package by the pulsar-client.

Reply:
It looks like this is causing the problem: https://github.com/apache/pulsar-client-python/commit/162afd53bda321992223cb209a509ee1c46f870a .

In Alpine, the grpcio version is 1.59.3 , https://pkgs.alpinelinux.org/package/v3.19/community/aarch64/py3-grpcio

@BewareMyPower Is there a way to allow using grpcio version 1.59.3 ?

Reply:
The operating system only requires the basic package, and we can use `pip` to install other dependencies, which are always the latest.

`apk` sometimes provides outdated packages.


>  The pip will uninstall the installed, and then to install a new grpcio package by the pulsar-client.

You can use the `RUN pip3 install --break-system-packages pulsar-client[all]==${PULSAR_CLIENT_PYTHON_VERSION} --verbose --prefer-binary` to reproduce this issue in the Dockerfile.

Reply:
> apk sometimes provides outdated packages.

I explained the root cause in the previous comment: https://github.com/apache/pulsar/pull/22613#discussion_r1583012275 .

Alpine comes with grpc 1.59.3 package. I don't see a reason why that couldn't be supported.